### PR TITLE
Share list through email

### LINKED ISSUE #16: As a user I want to be able to share my list with other users by email
8H

### PR DESCRIPTION
**Pull request by HEVpro**
+ Css modal style
+ Reorganize folders of the modal + imports
+ Rename modal files

### CODE REVIEW

File: backend/src/routers/list-router.js [hunk 1]
```diff
@@ -66,4 +67,20 @@ router.delete('/:listId/tasks', async (req, res) => {
...
+
+    res.status(200).end();
+}) 
+
+// Enpoint para los owners menos el logged
+router.get('/:listId/owners', async (req, res) => {
+    const allOwners = await listService.getListOwners(req.params.listId)
+    const otherOwners = await userService.getOwners(allOwners, req.userId);
+    res.json(otherOwners)
+})
```
Comment (vimtor) [inline backend/src/routers/list-router.js]: Move display logic to frontend not backend

File: backend/src/repositories/user-repository.js [hunk 2]
```diff
@@ -17,6 +18,14 @@ userSchema.pre("save", async function (next) {
 
 const userModel = model('User', userSchema)
 
+const getUser = async (id) => {
+  return userModel.find({_id: id});
+}
+
+const getAllUsers = async (id) => {
+  return userModel.find({_id: {$ne: id}});
+}
```
Comment (vimtor) [inline backend/src/repositories/user-repository.js]: We should validate the sharing in the backend not in the frontend.

File: backend/src/routers/users-router.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import userService from '../services/user-service'
+
+const router = Router()
+
+router.use(protect)
+
+router.get('/emails', async (req, res) => {
+    const users = await userService.getUsersEmails(req.userId); 
+    res.json(users);
+})
```
Comment (vimtor) [inline backend/src/routers/users-router.js]: We may not need this if email sharing validation is on backend.

File: backend/src/index.js [hunk 4]
```diff
@@ -20,6 +21,7 @@ app.use(rateLimit({windowMs: 60000, max: 100}))
 app.use(express.json())
 
 app.use(authRouter)
+app.use(usersRouter)
```
Comment (vimtor) [inline backend/src/index.js]: Missing `/users`

File: backend/src/routers/list-router.js (hunk 1 above)
Comment (vimtor) [reply backend/src/routers/list-router.js]: And return all owners

File: frontend/src/components/list-card/card-header/card-header.js [hunk 5]
```diff
@@ -1,34 +1,45 @@
...
+                    </svg>
+                    <svg xmlns="http://www.w3.org/2000/svg" onClick={openModal} style={{cursor: 'pointer'}} fill="gray" width="24" height="24" viewBox="0 0 24 24">
+                        <path
+                            d="M4.5 8.552c0 1.995 1.505 3.5 3.5 3.5s3.5-1.505 3.5-3.5-1.505-3.5-3.5-3.5S4.5 6.557 4.5 8.552zM19 8L17 8 17 11 14 11 14 13 17 13 17 16 19 16 19 13 22 13 22 11 19 11zM4 19h8 1 1v-1c0-2.757-2.243-5-5-5H7c-2.757 0-5 2.243-5 5v1h1H4z"/>
+                    </svg>
+                    <ShareModal
+                        modalIsOpen={modalIsOpen}
+                        toggleModal={setIsOpen}
+                    />
```
Comment (vimtor) [inline frontend/src/components/list-card/card-header/card-header.js]: Optionally you can create your own modal copying the styles from minimalist

image

**Review by vimtor — CHANGES_REQUESTED** [review]
Pretty good overall. Update the things I mentioned 😉

File: backend/src/repositories/user-repository.js (hunk 2 above)
Comment (HEVpro) [reply backend/src/repositories/user-repository.js]: Removed  this function in user-repository and the function to get all the emails of the app.

File: backend/src/index.js (hunk 4 above)
Comment (HEVpro) [reply backend/src/index.js]: Deleted my user-router, we are using only Francisco's router.

**Review by vimtor — APPROVED** [review]

**Pull request merged**